index.js

```diff
@@ -1,122 +1,110 @@
 const core = require('@actions/core');
 const shell = require('shelljs');
 
-async function bcc() {
-  core.info(`Start to build env`);
-  core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-  core.info(`Linux header`);
-  if (shell.exec("sudo apt-get install -y linux-headers-`uname -r`").code !== 0){
-    shell.echo("fail to install linux headers");
-    shell.exit(1);
-  }
-  let bcc_version = core.getInput('bcc_version');
-  core.debug(bcc_version);
-  if (bcc_version === undefined || bcc_version == null || bcc_version.length === 0) {
-    bcc_version="0.25.0";
-  }
-  core.info(`Get BCC with version:` + bcc_version);
-  if (shell.exec("wget https://github.com/sustainable-computing-io/kepler-ci-artifacts/releases/download/v"+bcc_version+"/bcc_v"+bcc_version+".tar.gz").code !==0){
-    shell.echo("fail to get BCC deb");
-    shell.exit(1);
-  }
-  if (shell.exec("tar -zxvf bcc_v"+bcc_version+".tar.gz").code !== 0) {
-    shell.echo("fail to untar BCC deb");
-    shell.exit(1);
+function getInputOrDefault(inputName, defaultValue) {
+  const input = core.getInput(inputName);
+  if (input === undefined || input == null || input.length === 0) {
+    return defaultValue;
   }
-  if (shell.exec("sudo dpkg -i libbcc_"+bcc_version+"-1_amd64.deb").code !== 0) {
-    shell.echo("fail to install BCC deb");
+  return input;
+}
+
+function executeCommand(command, errorMessage) {
+  const result = shell.exec(command);
+  if (result.code !== 0) {
+    shell.echo(errorMessage);
     shell.exit(1);
```

Comment:
maybe `shell.exit(result.code)` is better?

Reply:
this is to keep the consistency with previous code. If error occurs, the message is sufficient for debugging.

Reply:
for debugging usage, we can enable debugging flag at GHA web page UI... ok, let's keep current design.